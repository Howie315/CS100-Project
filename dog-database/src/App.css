@import url("https://fonts.googleapis.com/css2?family=Alata&display=swap");
.App {
  text-align: center;
}
.ui-form {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
}
.App-logo {
  height: 40vmin;
  pointer-events: none;
}
.img-with-text {
  text-align: center;

  width: [width of img];
}
.img-with-text img {
  display: block;
  margin: 0 auto;
}

@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-spin infinite 20s linear;
  }
}
input[type="text"]:focus {
  border: 2px solid rgb(0, 236, 236);
}

input[type="text"] {
  margin-bottom: 2rem;
  padding: 5px;
  border: 2px solid gray;
  outline: none;
  border-radius: 5px;
  width: 30vw;
}
input[type="submit"] {
  padding: 0.5rem;
  width: 30vw;
  cursor: pointer;
  background: linear-gradient(to right, rgb(0, 238, 238), rgb(0, 190, 204));
  color: white;
  font-weight: 700;
  font-size: 1.2rem;
  border: none;
  outline: none;
  border-radius: 5px;
  transition: all 0.2s ease-in;
}

input[type="submit"]:hover {
  background: linear-gradient(to right, rgb(0, 168, 197), rgb(2, 98, 224));
}
.results-add {
  height: 100vh;
  width: 100%;
  display: flex;
  align-items: center;
  justify-content: center;
  flex-direction: column;
  background: rgb(10, 10, 10);
}
.App-header {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: top;
  justify-content: top;
  font-size: calc(13px + 2vmin);
  color: white;
  text-align: center;
}

.App-link {
  color: #61dafb;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}
