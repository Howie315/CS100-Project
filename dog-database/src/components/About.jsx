import React from "react";

function About() {
  return (
    <div className="about">
      <div class="container">
        <div class="row align-items-center my-5">
          <div class="col-lg-7">
            <img
              class="img-fluid rounded mb-4 mb-lg-0"
              src="http://placehold.it/900x400"
              alt=""
            />
            <img src = {require("./dogadopt.png")}
            width = "575"
            height = "550" />
          </div>
          <div class="col-lg-5">
            <h1 class="font-weight-normal-bold">About</h1>
            <p>
             Dog DataBase is an online dog center where users can access and and add, remove, or learn about a dog. We are driven by the growing number of dogs 
             occupying animal shelters and wish to find homes for our canine companions. Here, you can browse and pick from a variety of dogs to adopt. 
            </p>
          </div>
        </div>
      </div>
    </div>
  );
}

export default About;