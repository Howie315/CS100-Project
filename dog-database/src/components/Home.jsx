import React from "react";

function Home() {
  return (
    <div className="home">
      <div class="container">
        <div class="row align-items-center my-5">
          <div class="col-lg-11">
          <h1 class="font-weight-normal-bold"
          style = {{ color: "black" }}
          > Home</h1>
            <p>
              Welcome to Dog DataBase, the site to browse your furry friends! 
            </p>
            <img
              class="img-fluid rounded mb-4 mb-lg-0"
             src = {require("./icon.png")}
              width = "575"
              height = "550"
              alt=""
            />
          
      
          </div>
        </div>
      </div>
    </div>
  );
}
const StyleButton = styled.a`
display:block;
padding-left:0%;
button {
    position: relative;
    //display:block;
    height: 45px;
    width: 150px;
    margin: 10px 7px;
    padding: 5px 5px;
    font-weight: 700;
    font-size: 20px;
    letter-spacing: 2px;
    color: #383736;
    border: 2px #383736 solid;
    border-radius: 4px;
    text-transform: uppercase;
    outline: 0;
    overflow:hidden;
    background: white;
    color: black;
    z-index: 1;
    cursor: pointer;
    transition:         0.08s ease-in;
    -o-transition:      0.08s ease-in;
    -ms-transition:     0.08s ease-in;
    -moz-transition:    0.08s ease-in;
    -webkit-transition: 0.08s ease-in;
}

`;

export default Home;