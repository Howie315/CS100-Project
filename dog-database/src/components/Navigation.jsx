import React from "react";
import { NavLink } from "react-router-dom";
import styled from 'styled-components';


function Navigation() {
  return (
    <div className="navigation">
      <nav className="navbar navbar-expand navbar-dark bg-dark">
    
            <img src = {require("./download.png")}
            //width = "70"
            height = "70"/>
        <div className="container">
          
          <NavLink className="navbar-brand" to="/">
            <h1 class = "font-weight-normal-bold">Dog Database</h1>
          </NavLink>
          
          <div>
            <ul className="navbar-nav ml-auto">
              <li className="nav-item">
                <NavLink className="nav-link" to="/">
                <h1 class = "font-weight-normal-bold">| Home |</h1>
                  <span className="sr-only">(current)</span>
                </NavLink>
                
              </li>
              
              <li className="nav-item">
                <NavLink className="nav-link" to="/about">
                <h1 class = "font-weight-normal-bold">About |</h1>
                </NavLink>
              </li>
              <li className="nav-item">
                <NavLink className="nav-link" to="/DogListings">
                <h1 class = "font-weight-normal-bold">DogListings |</h1>
                </NavLink>
              </li>
              <li className="nav-item">
                <NavLink className="nav-link" to="/AddDog">
                <h1 class = "font-weight-normal-bold">Dog Add |</h1>
                </NavLink> 
              </li>
              <li className="nav-item">
                <NavLink className="nav-link" to="/RemoveDog">
                <h1 class = "font-weight-normal-bold">Dog Remove |</h1>
                </NavLink>
              </li>
            </ul>
          </div>
        </div>
       
            <img src = {require("./download.png")}
            //width = "70"
            height = "70"/>
      </nav>
    </div>
  );
}

export default Navigation;