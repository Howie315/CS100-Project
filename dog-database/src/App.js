import React from "react";
import "./App.css";
import Loading from "./loading.js";
import { BrowserRouter as Router, Route, Routes } from "react-router-dom";
import Navigation from "./components/Navigation";
import Home from "./components/Home";
import About from "./components/About";
import DogListings from "./components/DogListings";
import AddDog from "./components/AddDog";
import RemoveDog from "./components/RemoveDog";

function App() {
  return (
	
 <div className="App">
<header className="App-header">
	<Loading/>
</header>
</div>





   
  );
}

export default App;




