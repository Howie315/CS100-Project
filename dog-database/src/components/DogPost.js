import React, { Component } from "react";
import axios, { post } from "axios";
import PostTile from "./PostTile";
import styled from "styled-components";
import Post from "./Post";
import instance from "../axios-conn";
import { storage, app } from "../firebase";
import { trackPromise } from "react-promise-tracker";
import { toast } from "react-toastify";

class DogPost extends Component {
  state = {
    id: null,
    theName: null,
    age: null,
    gender: null,
    breed: null,
    description: null,
    file: null,
    dogTagList: [],
  };

  componentDidMount() {
    trackPromise(
      instance.get("/posts.json").then((res) => {
        console.log(res.data);
        const result = [];
        for (let key in res.data) {
          result.push({
            ...res.data[key],
            id: key,
          });
        }
        this.setState({ dogTagList: result });
      })
    );

    //.catch((err) => console.log(err));
  }

  onFormSubmit(event) {
    event.preventDefault();
    this.fileUpload(this.state.file).then((response) => {
      console.log(response.data);
    });
  }

  onChange(event) {
    this.setState({ file: event.target.files[0] });
  }
  fileUpload(file) {
    const formData = new FormData();
    formData.append("file", file);
    const config = {
      headers: {
        "content-type": "multipart/form-data",
      },
    };
    return post("gs://dog-database-d87bb.appspot.com", formData, config);
  }

  handleChange = (event) => {
    this.setState({ [event.target.name]: event.target.value });
  };

  handleSubmit = (event) => {
    event.preventDefault();

    const data = {
      theName: this.state.theName,
      age: this.state.age,
      gender: this.state.gender,
      breed: this.state.breed,
      description: this.state.description,
      file: this.state.file,
      id: this.state.id,
    };
    console.log(data);

    instance
      .post("/posts.json", data)
      .then((res) => {
        const fresh_post = res.data;
        this.state.dogTagList.push(fresh_post);
      })
      .catch((err) => console.log(err));
  };

  render() {
    return (
      <div>
        <StyledCreatePost>
          <form onSubmit={this.onFormSubmit}>
            <label>Image Upload</label>
            <input type="file" onChange={this.onChange} />
            <button className="arrow" type="submit">
              upload
            </button>
          </form>
          <form
            className="ui-form"
            autoComplete="off"
            onSubmit={this.handleSubmit}
          >
            <label>Name</label>
            <input
              autofocus
              type="text"
              placeholder="name.."
              name="theName"
              onChange={this.handleChange}
            />
            <label>Description</label>
            <input
              type="text"
              placeholder="contents.."
              name="description"
              onChange={this.handleChange}
            />
            <label>Age</label>
            <input
              type="text"
              placeholder="age in years.."
              name="age"
              onChange={this.handleChange}
            />
            <label>Gender</label>
            <input
              type="text"
              placeholder="gender.."
              name="gender"
              onChange={this.handleChange}
            />
            <label>Breed</label>
            <input
              type="text"
              placeholder="breed.."
              name="breed"
              onChange={this.handleChange}
            />

            <button className="arrow" type="submit">
              SUBMIT
            </button>
          </form>
        </StyledCreatePost>
        <StyledPost>
          {this.state.dogTagList &&
            this.state.dogTagList.map((dog) => (
              <PostTile
                key={dog.id}
                id={dog.id}
                theName={dog.theName}
                age={dog.age}
                gender={dog.gender}
                breed={dog.breed}
                description={dog.description}
                file={dog.file}
              />
            ))}
        </StyledPost>
      </div>
    );
  }
}

const StyledCreatePost = styled.a`
  //display: block;
  padding-left: 5%;
  font-size: 25px;
  button {
    position: relative;
    //display: block;
    height: 35px;
    width: 100px;
    margin: 10px 7px;
    padding: 5px 5px;
    font-weight: 700;
    font-size: 15px;
    letter-spacing: 2px;
    color: white;
    border: 2px white solid;
    border-radius: 4px;
    text-transform: uppercase;
    outline: 0;
    overflow: hidden;
    background: none;
    z-index: 1;
    cursor: pointer;
    transition: 0.08s ease-in;
    -o-transition: 0.08s ease-in;
    -ms-transition: 0.08s ease-in;
    -moz-transition: 0.08s ease-in;
    -webkit-transition: 0.08s ease-in;
  }
  .arrow:hover:before {
    left: 84%auto;
    opacity: 1;
  }
  .arrow:hover {
    width: 170px;
  }
  .arrow:before {
    content: "â†’";
    position: absolute;
    color: 383736;
    left: 83%;
    opacity: 0;
    -webkit-transition: all 250mx cubic-bezier(0.68, -0.55, 0.265, 1.55);
  }
`;

const StyledPost = styled.a`
  display: block;
  padding-left: -5%;
  a {
    color: black;
    text-decoration: none;
  }
`;

export default DogPost;
