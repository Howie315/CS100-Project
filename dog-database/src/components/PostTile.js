import axios from "axios";
import React, { Component, useEffect, useState } from "react";
import { Link } from "react-router-dom";
import styled from "styled-components";
import instance from "../axios-conn";
import DogPost from "./DogPost";
import { storage, app } from "../firebase";

class PostTile extends Component {
  handleRemove = (id) => {
    instance.delete(`posts/${id}.json`).then((res) => {
      console.log(res);
      console.log(res.data);
    });

    //     .catch((err) => console.log(err));
  };

  render() {
    return (
      <StyledPostTile>
        <h1>{this.props.theName}</h1>
        <div class="container">
          <div class="row">
            <div class="col-md-5 img img-3 d-flex justify-content-center align-items-center">
              <div class="img-with-text">
                <img src={this.props.file} width="500" height="500" />
                <font size="6">
                  <p>{this.props.breed}</p>
                  <p>{this.props.gender}</p>
                  <p>{this.props.age}</p>
                </font>
              </div>
            </div>

            <div class="col-md-7 wrap-about pl-md-5 ">
              <div class="heading-section">
                <h2 class="mb-4">Dog</h2>
              </div>
              <font size="6">
                <p>{this.props.description}</p>
              </font>
            </div>
          </div>
        </div>

        <div class="col-lg-7">{this.props.file}</div>

        <form onSubmit={this.handleRemove}>
          <button className="arrow" type="submit">
            DELETE
          </button>
        </form>
      </StyledPostTile>
    );
  }
}

const StyledPostTile = styled.a`
  display: block;
  button {
    position: relative;
    //display:block;
    height: 35px;
    width: 100px;
    margin: 10px 7px;
    padding: 5px 5px;
    font-weight: 700;
    font-size: 15px;
    letter-spacing: 3px;
    color: #383736;
    border: 2px #383736 solid;
    border-radius: 4px;
    text-transform: uppercase;
    outline: 0;
    overflow: hidden;
    background: none;
    z-index: 1;
    cursor: pointer;
    transition: 0.08s ease-in;
    -o-transition: 0.08s ease-in;
    -ms-transition: 0.08s ease-in;
    -moz-transition: 0.08s ease-in;
    -webkit-transition: 0.08s ease-in;
  }
  .arrow:hover:before {
    left: 84%auto;
    opacity: 1;
  }
  .arrow:hover {
    width: 170px;
  }
  .arrow:before {
    content: "â†’";
    position: absolute;
    color: 383736;
    left: 83%;
    opacity: 0;
    -webkit-transition: all 250mx cubic-bezier(0.68, -0.55, 0.265, 1.55);
  }
`;

export default PostTile;
