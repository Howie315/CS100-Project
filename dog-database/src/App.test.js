import { render, screen } from "@testing-library/react";
import App from "./App";

import DogPost from "./components/DogPost";
import PostTile from "./components/PostTile";
import renderer from "react-test-renderer";
import Navigation from "./components/Navigation";
import About from "./components/About";
import Home from "./components/Home";
it("renders correctly across screens", () => {
  const tree = renderer.create(<DogPost />).toJSON();
  expect(tree).toMatchSnapshot();
});

test("renders correctly", () => {
  const tree = renderer.create(<PostTile />).toJSON();
  expect(tree).toMatchSnapshot();
});
