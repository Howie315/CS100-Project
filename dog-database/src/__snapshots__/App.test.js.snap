// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<a
  className="sc-bdvvtL bgTaay"
>
  <h1 />
  <div
    class="container"
  >
    <div
      class="row"
    >
      <div
        class="col-md-5 img img-3 d-flex justify-content-center align-items-center"
      >
        <div
          class="img-with-text"
        >
          <img
            height="500"
            src="dogbone.png"
            width="500"
          />
          <font
            size="6"
          >
            <p />
            <p />
            <p />
          </font>
        </div>
      </div>
      <div
        class="col-md-7 wrap-about pl-md-5 "
      >
        <div
          class="heading-section"
        >
          <h2
            class="mb-4"
          >
            Dog
          </h2>
        </div>
        <font
          size="6"
        >
          <p />
        </font>
      </div>
    </div>
  </div>
  <div
    class="col-lg-7"
  />
  <form
    onSubmit={[Function]}
  >
    <button
      className="arrow"
      type="submit"
    >
      DELETE
    </button>
  </form>
</a>
`;

exports[`renders correctly 2`] = `
<a
  className="sc-bdvvtL bgTaay"
>
  <h1 />
  <div
    class="container"
  >
    <div
      class="row"
    >
      <div
        class="col-md-5 img img-3 d-flex justify-content-center align-items-center"
      >
        <div
          class="img-with-text"
        >
          <img
            height="500"
            src="dogbone.png"
            width="500"
          />
          <font
            size="6"
          >
            <p />
            <p />
            <p />
          </font>
        </div>
      </div>
      <div
        class="col-md-7 wrap-about pl-md-5 "
      >
        <div
          class="heading-section"
        >
          <h2
            class="mb-4"
          >
            Dog
          </h2>
        </div>
        <font
          size="6"
        >
          <p />
        </font>
      </div>
    </div>
  </div>
  <div
    class="col-lg-7"
  />
  <form
    onSubmit={[Function]}
  >
    <button
      className="arrow"
      type="submit"
    >
      DELETE
    </button>
  </form>
</a>
`;

exports[`renders correctly 3`] = `
<div>
  <a
    className="sc-dkPtRN dzyVLj"
  >
    <form
      onSubmit={[Function]}
    >
      <label>
        Image Upload
      </label>
      <br />
      <input
        onChange={[Function]}
        type="file"
      />
      <button
        className="arrow"
        onclick={[Function]}
        type="submit"
      >
        Upload
      </button>
      <br />
    </form>
    <form
      autoComplete="off"
      className="ui-form"
      onSubmit={[Function]}
    >
      <label>
        Name
      </label>
      <input
        autofocus={true}
        name="theName"
        onChange={[Function]}
        placeholder="name.."
        type="text"
      />
      <label>
        Description
      </label>
      <input
        name="description"
        onChange={[Function]}
        placeholder="contents.."
        type="text"
      />
      <label>
        Age
      </label>
      <input
        name="age"
        onChange={[Function]}
        placeholder="age in years.."
        type="text"
      />
      <label>
        Gender
      </label>
      <input
        name="gender"
        onChange={[Function]}
        placeholder="gender.."
        type="text"
      />
      <label>
        Breed
      </label>
      <input
        name="breed"
        onChange={[Function]}
        placeholder="breed.."
        type="text"
      />
      <button
        className="arrow"
        type="submit"
      >
        SUBMIT
      </button>
    </form>
  </a>
  <a
    className="sc-hKwDye cSgGvt"
  />
</div>
`;

exports[`renders correctly across screens 1`] = `
<div>
  <a
    className="sc-dkPtRN dzyVLj"
  >
    <form
      onSubmit={[Function]}
    >
      <label>
        Image Upload
      </label>
      <br />
      <input
        onChange={[Function]}
        type="file"
      />
      <button
        className="arrow"
        onclick={[Function]}
        type="submit"
      >
        Upload
      </button>
      <br />
    </form>
    <form
      autoComplete="off"
      className="ui-form"
      onSubmit={[Function]}
    >
      <label>
        Name
      </label>
      <input
        autofocus={true}
        name="theName"
        onChange={[Function]}
        placeholder="name.."
        type="text"
      />
      <label>
        Description
      </label>
      <input
        name="description"
        onChange={[Function]}
        placeholder="contents.."
        type="text"
      />
      <label>
        Age
      </label>
      <input
        name="age"
        onChange={[Function]}
        placeholder="age in years.."
        type="text"
      />
      <label>
        Gender
      </label>
      <input
        name="gender"
        onChange={[Function]}
        placeholder="gender.."
        type="text"
      />
      <label>
        Breed
      </label>
      <input
        name="breed"
        onChange={[Function]}
        placeholder="breed.."
        type="text"
      />
      <button
        className="arrow"
        type="submit"
      >
        SUBMIT
      </button>
    </form>
  </a>
  <a
    className="sc-hKwDye cSgGvt"
  />
</div>
`;

exports[`renders correctly across screens 2`] = `
<div>
  <a
    className="sc-dkPtRN dzyVLj"
  >
    <form
      onSubmit={[Function]}
    >
      <label>
        Image Upload
      </label>
      <br />
      <input
        onChange={[Function]}
        type="file"
      />
      <button
        className="arrow"
        onclick={[Function]}
        type="submit"
      >
        Upload
      </button>
      <br />
    </form>
    <form
      autoComplete="off"
      className="ui-form"
      onSubmit={[Function]}
    >
      <label>
        Name
      </label>
      <input
        autofocus={true}
        name="theName"
        onChange={[Function]}
        placeholder="name.."
        type="text"
      />
      <label>
        Description
      </label>
      <input
        name="description"
        onChange={[Function]}
        placeholder="contents.."
        type="text"
      />
      <label>
        Age
      </label>
      <input
        name="age"
        onChange={[Function]}
        placeholder="age in years.."
        type="text"
      />
      <label>
        Gender
      </label>
      <input
        name="gender"
        onChange={[Function]}
        placeholder="gender.."
        type="text"
      />
      <label>
        Breed
      </label>
      <input
        name="breed"
        onChange={[Function]}
        placeholder="breed.."
        type="text"
      />
      <button
        className="arrow"
        type="submit"
      >
        SUBMIT
      </button>
    </form>
  </a>
  <a
    className="sc-hKwDye cSgGvt"
  />
</div>
`;
